packages/micro/src/lib/index.ts

```diff
@@ -0,0 +1,215 @@
+// Native
+import { Stream, Readable } from 'stream';
+// Packages
+import contentType from 'content-type';
+import getRawBody from 'raw-body';
+import type { RawBodyError } from 'raw-body';
+//Types
+import type { IncomingMessage, ServerResponse, RequestListener } from 'http';
+
+// slight modification of is-stream https://github.com/sindresorhus/is-stream/blob/c918e3795ea2451b5265f331a00fb6a8aaa27816/license
+function isStream(stream: unknown): stream is Stream {
+  return (
+    stream !== null &&
+    typeof stream === 'object' &&
+    stream instanceof Stream &&
+    typeof stream.pipe === 'function'
+  );
+}
+
+function readable(stream: unknown): stream is Readable {
+  return (
+    isStream(stream) && // TODO: maybe this isn't needed because we could use only the checks below
+    stream instanceof Readable &&
+    stream.readable
+  );
+}
+
+export type RequestHandler = (
+  req: IncomingMessage,
+  res: ServerResponse,
+) => unknown;
+
+type Serve = (fn: RequestHandler) => RequestListener;
+
+export const serve: Serve = (fn) => (req, res) => run(req, res, fn);
+
+export class HttpError extends Error {
+  constructor(message: string) {
+    super(message);
+    Object.setPrototypeOf(this, HttpError.prototype);
+  }
+
+  statusCode?: number;
+  originalError?: Error;
+}
+
+function isError(error: unknown): error is Error | HttpError {
+  return error instanceof Error || error instanceof HttpError;
+}
+
+export const createError = (code: number, message: string, original: Error) => {
+  const err = new HttpError(message);
+
+  err.statusCode = code;
+  err.originalError = original;
+
+  return err;
+};
+
+export const send = (
+  res: ServerResponse,
+  code: number,
+  obj: unknown = null,
+) => {
+  res.statusCode = code;
+
+  if (obj === null) {
+    res.end();
+    return;
+  }
+
+  if (Buffer.isBuffer(obj)) {
+    if (!res.getHeader('Content-Type')) {
+      res.setHeader('Content-Type', 'application/octet-stream');
+    }
+
+    res.setHeader('Content-Length', obj.length);
+    res.end(obj);
+    return;
+  }
+
+  if (obj instanceof Stream || readable(obj)) {
+    //TODO: Wouldn't (obj instanceof Stream) be the only check here? Do we specifically need a Readable stream or a Stream object that's not of NodeJS Stream?
```

Comment:
I'm not sure about this. The comment is for someone here to let me know why we do this. Or discuss later in the future if this is necessary.